docs/go/how-to-use-a-side-effect-in-go.md

```diff
@@ -0,0 +1,47 @@
+---
+id: how-to-use-a-side-effect-in-go
```

Comment:
Were you aware of the existing file for this?
https://github.com/temporalio/documentation/blob/master/docs/go/how-to-execute-a-side-effect-in-go.md

tbh - I think the existing one is more to the point and clearer atm